src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java

```diff
@@ -32,6 +39,20 @@
 
 public class ArchUnitTest {
 
+    /**
+     * List of classes that don't follow the rule
+     * {@code testClassesInApiDoNotDependOnClassesInUtil}.
+     */
+    private static final Set<String> API_CLASSES_DEPENDENT_ON_UTILS = Set.of(
```

Comment:
@Vyom-Yadav do you think it would be better to use `Set<Class<?>>`? This might help us to find problems in the future if we remove/ move/ change a class.

Reply:
There were two reasons why I didn't go for that:

1. It will add a lot of imports if the suppression list is long.
2. Not all classes are public, we won't be able to import them.
(ex - `com.puppycrawl.tools.checkstyle.api.AutomaticBean$PatternConverter`)

Reply:
Both good points, agreed.